### DISCUSSION TITLE
Run on start and schedule

### DISCUSSION THREAD

**Discussion by rodrigost23** [containrrr/watchtower#1975] [Q&A]
I know there is `--run-once`, but using this flag, the container exits afterwards. Using only `--interval 86400` makes it wait for 24 hours for the **first** check.

Is there a way to use `--interval` or `--schedule` but do the first check right away? The problem is that my server is not always turned on, and all services are `restart: unless-stopped` in docker-compose, so when watchtower starts, it always schedules for the next day (23 hours, 59 minutes, 59 seconds).

**Comment by avtar** [comment]
Has anyone figured out a way to achieve this?

**Reply by rodrigost23** [reply]
I ended up creating a custom Dockerfile:
```dockerfile
FROM containrrr/watchtower AS watchtower

FROM alpine

COPY --from=watchtower /watchtower /

RUN echo '/watchtower "$@" --run-once && /watchtower "$@"' > /entrypoint.sh

HEALTHCHECK CMD [ "/watchtower", "--health-check"]

ENTRYPOINT [ "sh", "/entrypoint.sh" ]
```

**Reply by avtar** [reply]
Nice workaround! Thanks 🙏
